backend/fms_core/migrations/0018_test4.py

```diff
@@ -0,0 +1,70 @@
+# Generated by Django 3.1 on 2021-03-03 19:08
+
+from django.db import migrations, models, transaction
+import django.db.models.deletion
+import json
+from datetime import datetime, timedelta
+
+from ..utils import float_to_decimal
+
+
+def move_volume_history_to_update_process(apps, schema_editor):
+    Sample = apps.get_model("fms_core", "Sample")
+    Version = apps.get_model("reversion", "Version")
+    Revision = apps.get_model("reversion", "Revision")
+    Protocol = apps.get_model("fms_core", "Protocol")
+    Process = apps.get_model("fms_core", "Process")
+    ProcessSample = apps.get_model("fms_core", "ProcessSample")
+
+    update_protocol = Protocol.objects.get(name="Update")
+
+    for sample in Sample.objects.all():
+        sample_versions = Version.objects.filter(content_type__model="sample", object_id=sample.id).order_by('id')
+        previous_update_comment = ''
+        previous_volume_value = None
+        for index, sample_version in enumerate(sample_versions):
+            data = json.loads(sample_version.serialized_data)
+            update_comment = data[0]["fields"]["update_comment"]
+            if index != 0: # skip first version, as this will be an object creation
+                volume_history = data[0]["fields"]["volume_history"]
+                last_vh = volume_history[-1]
+                if last_vh["update_type"] == "update":
+                    process_sample_comment = ''
+                    if update_comment != previous_update_comment:
```

Comment:
What happens if someone submit 2 updates with the same comment back to back ?

Reply:
I dont think there is a way around it... and Id rather not have the information displayed directly (remember, it is still stored in versions) than having the wrong information being displayed as an update comment (i.e. having as an update comment: 'extraction done blablabla' for a sample update)

Reply:
Yeah this is all right ;)  i don't see a way around it either ...